wp-dependency-installer.php

```diff
@@ -585,39 +614,42 @@ public function admin_notices() {
 				return false;
 			}
 			foreach ( $this->notices as $notice ) {
-				$status  = empty( $notice['status'] ) ? 'updated' : $notice['status'];
-				$message = empty( $notice['message'] ) ? '' : esc_html( $notice['message'] );
-
-				if ( ! empty( $notice['action'] ) ) {
-					$action   = esc_attr( $notice['action'] );
-					$message .= ' <a href="javascript:;" class="wpdi-button" data-action="' . $action . '" data-slug="' . $notice['slug'] . '">' . ucfirst( $action ) . ' Now &raquo;</a>';
+				$status      = isset( $notice['status'] ) ? $notice['status'] : 'updated';
+				$source      = isset( $notice['source'] ) ? $notice['source'] : __( 'Dependency' );
+				$class       = esc_attr( $status ) . ' notice is-dismissible dependency-installer';
+				$label       = esc_html( $this->get_dismiss_label( $source ) );
+				$message     = '';
+				$action      = '';
+				$dismissible = '';
+
+				if ( isset( $notice['message'] ) ) {
+					$message = esc_html( $notice['message'] );
 				}
-
-				/**
-				 * Filters the dismissal timeout.
-				 *
-				 * @since 1.4.1
-				 *
-				 * @param string|int '7'           Default dismissal in days.
-				 * @param  string     $notice['source'] Plugin slug of calling plugin.
-				 * @return string|int Dismissal timeout in days.
-				 */
-				$timeout     = '-' . apply_filters( 'wp_dependency_timeout', '7', $notice['source'] );
-				$dismissible = isset( $notice['slug'] )
-				? 'dependency-installer-' . dirname( $notice['slug'] ) . $timeout
-				: null;
-				if ( class_exists( '\PAnd' ) && ! \PAnD::is_admin_notice_active( $dismissible ) ) {
-					continue;
+				if ( isset( $notice['action'] ) ) {
+					$action = sprintf(
+						' <a href="javascript:;" class="wpdi-button" data-action="%1$s" data-slug="%2$s">%3$s Now &raquo;</a> ',
+						esc_attr( $notice['action'] ),
+						esc_attr( $notice['slug'] ),
+						esc_html( ucfirst( $notice['action'] ) )
+					);
+				}
+				if ( isset( $notice['slug'] ) ) {
+					/**
+					 * Filters the dismissal timeout.
+					 *
+					 * @since 1.4.1
+					 *
+					 * @param string|int '7'           Default dismissal in days.
+					 * @param  string     $notice['source'] Plugin slug of calling plugin.
+					 * @return string|int Dismissal timeout in days.
+					 */
+					$timeout     = apply_filters( 'wp_dependency_timeout', '7', $source );
+					$dependency  = dirname( $notice['slug'] );
+					$dismissible = empty( $timeout ) ? '' : sprintf( 'dependency-installer-%1$s-%2$s', esc_attr( $dependency ), esc_attr( $timeout ) );
```

Comment:
In this way, this code works correctly:

```php
add_filter(
  'wp_dependency_timeout',
  function( $timeout, $source ) {
    return 0; // <-- i.e. un-dismissible notices
  }, 10, 2
);
```

Reply:
If you want to override the minimum value in PAnD the PR should be there.